src/dataflow/src/render/reduce.rs

```diff
@@ -1063,6 +1066,135 @@ where
         })
 }
 
+/// Accumulates values for the various types of accumulable aggregations.
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Serialize, Deserialize)]
+enum Accumulator {
+    /// Accumulates boolean values.
+    Bool {
+        /// The number of `true` values observed.
+        trues: isize,
+        /// The number of `false` values observed.
+        falses: isize,
+        /// The total number of values observed.
+        total: isize,
+    },
+    /// Accumulates the count of non-NULL values.
+    Count {
+        /// The number of non-NULL values observed.
+        non_nulls: isize,
+        /// The total number of values observed.
+        ///
+        /// This value is not actually observed directly by the code in this
+        /// module, but it serves to ensure that `is_zero()` returns false when
+        /// at least one non-NULL value has been observed. This is required to
+        /// produce an explicit `0` when a group contains zero elements;
+        /// otherwise differential would produce no results at all.
+        total: isize,
+    },
+    /// Accumulates simple numeric values.
+    SimpleNumber {
+        /// The sum of all non-NULL values observed.
+        sum: i128,
+        /// The number of non-NULL values observed.
+        count: isize,
+    },
```

Comment:
Does this need a `total` field? I'm just going off of the comments, but if only non-null things contribute, we'll miss null values going in to a sum and produce nothing rather than zero.

Reply:
Also, as they have similar sizes, would it make sense to unify this with `Count`?

Reply:
Reading ahead, it looks like `Datum::Null` maps to a `(0, 0)` entry for sums, which seems like it runs the risk of not producing an output for a `Sum(Null)` aggregation.

Reply:
Indeed. Fixed to track both `non_nulls` and `total`, as was done before.